node/connection.js

```diff
@@ -251,7 +251,10 @@ TChannelConnection.prototype.onWriteError = function onWriteError(err) {
 
 TChannelConnection.prototype.onHandlerError = function onHandlerError(err) {
     var self = this;
-    self.resetAll(err);
+
+    if (err) {
```

Comment:
it was passed as a callback in places; so this method is a callback and thus err is optional